### ISSUE TITLE
Bug: pagy_search wrong number of arguments

### ISSUE THREAD

**Issue by Frexuz** [ddnexus/pagy#640]
### Before submitting...

- [X] I upgraded to the latest version of pagy
- [X] I searched through the [Documentation](https://ddnexus.github.io/pagy/)
- [X] I searched through the [Issues](https://github.com/ddnexus/pagy/issues)
- [X] I searched through the [Q&A](https://github.com/ddnexus/pagy/discussions/categories/q-a)

### I am providing...

- [ ] Simple step by step list that would work in IRB with the [Pagy::Console](https://ddnexus.github.io/pagy/docs/api/console)

- [ ] Plain ruby file that can run as `ruby my-problem.rb`

- [ ] Edited copy of the single file [pagy_standalone_app.ru](https://github.com/ddnexus/pagy/blob/master/apps/pagy_standalone_app.ru)

- [x] Edited copy of the single file [pagy_bare_rails.rb](https://github.com/ddnexus/pagy/blob/master/apps/pagy_bare_rails.rb)

- [ ] Link of my own branch forked from pagy, which contains an added test file

- [ ] Link of my own branch forked from [any of the rails apps listed here.](https://github.com/stars/benkoshy/lists/rails-demo-apps-for-pagy)

- [ ] A `docker-compose` file that uses only docker images and no local context and runs with `docker-compose up`

### Description

Search with meilisearch with filters, get correct results
```
irb(main):009:0> Profile.search('', { filter: [['interest_ids = 6']] })
  Profile Load (7.2ms)  SELECT "profiles".* FROM "profiles" WHERE "profiles"."id" IN ($1, $2, $3, $4, $5, $6, $7)  [["id", 9], ["id", 23], ["id", 33], ["id", 38], ["id", 53], ["id", 55], ["id", 69]]
=>
[#<Profile:0x00000001250df9f0
  id: 9,
  ....
```
replace `search` with `pagy_search`:
```
rb(main):012:0> Profile.pagy_search('', { filter: ['interest_ids = 6'] })
/Users/frexuz/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/pagy-6.4.3/lib/pagy/extras/meilisearch.rb:13:in `pagy_meilisearch': wrong number of arguments (given 2, expected 0..1) (ArgumentError)
irb(main):013:0>
```
<img width="538" alt="image" src="https://github.com/ddnexus/pagy/assets/49692/8db77a53-9f59-44c1-a632-a0bb0fa09039">

--------------------

Updating with base rails debug file:

```
require 'bundler/inline'

gemfile(true) do
  source 'https://rubygems.org'

  # Debuggers
  gem 'debug'

  gem 'pagy', '6.4.3'
  gem 'stringio', '3.1.0'

  gem 'rails', '7.0.8'
  gem 'actionpack'
  gem 'railties'
  gem 'sqlite3'


  ## Optional: Meilisearch example
  gem 'meilisearch-rails', '~> 0.10.2'
end

require 'debug'

## Pagy Set-up
require 'pagy/extras/meilisearch'
require 'pagy/extras/metadata'
require 'pagy/extras/overflow'
require 'pagy/extras/trim'

Pagy::DEFAULT[:trim_extra] = false
Pagy::DEFAULT[:items] = 20
Pagy::DEFAULT[:metadata] = %i[first_url last_url]
Pagy::DEFAULT[:overflow] = :empty_page

# Rails set up:
require 'action_controller/railtie'
require 'active_record'
require 'minitest/autorun' # runs tests automatically

class TestApp < Rails::Application # :nodoc:
  config.root = __dir__
  config.hosts << 'www.example.com'
  config.hosts << 'example.org'
  config.session_store :cookie_store, key: 'cookie_store_key'
  secrets.secret_key_base = 'secret_key_base'

  config.logger = Logger.new($stdout)
  Rails.logger  = config.logger

  routes.draw do
    get '/' => 'test#index', as: 'test'
  end

  ## Optional: Meilisearch example
  MeiliSearch::Rails.configuration = {
    meilisearch_host: 'http://localhost:7700',
    meilisearch_api_key: 'password'
  }
end

ActiveRecord::Base.establish_connection(adapter: 'sqlite3', database: ':memory:')

ActiveRecord::Base.connection.create_table(:books) do |t|
  t.text :name
  t.integer :sold
end

ActiveRecord::Base.logger = Logger.new($stdout)

class Book < ActiveRecord::Base # :nodoc:
  belongs_to :author

  ## Optional: Meilisearch example
  include MeiliSearch::Rails
  extend Pagy::Meilisearch

  meilisearch do
    attribute :name
    attribute :sold
  end
end # :nodoc:

1.upto(11) do |i|
  Book.create(name: "Book ##{i}", sold: i)
end

## Optional: Meilisearch example
Book.reindex!

books = Book.pagy_search('*', { filter: 'sold > 5' })
@pagy, @books = pagy_meilisearch(books, items: 10)
```

**Comment by ddnexus** [comment]
Hi @Frexuz , the  "Simple step by step list that would work in IRB with the Pagy::Console" should not include calling your own app, which code is not accessible to me. Indeed that support is for bugs that are intrinsic to pagy itself, i.e. don't require other code in order to be reproduced.

Please, post any other support that will be convenient for you, but would not run code that is not provided/included.

Thank you.

**Comment by ddnexus** [comment]
BTW, I will make that support description more explicit because it's admittedly ambiguous. Sorry for that.

**Comment by ddnexus** [comment]
@Frexuz 

I've just realized that you pass different arguments to `Profile.search` and `Profile.pagy_search`.

I bet that that's the "bug" 🙂

**Comment by Frexuz** [comment]
> @Frexuz
> 
> I've just realized that you pass different arguments to `Profile.search` and `Profile.pagy_search`.
> 
> I bet that that's the "bug" 🙂

if the error says `/pagy-6.4.3/lib/pagy/extras/meilisearch.rb:13:in `pagy_meilisearch': wrong number of arguments (given 2, expected 0..1) (ArgumentError)` , the library clearly doesnt take 2 arguments for some reason. I don't see how thats related to application code?

The different filters passed was a typo, but here:

Profile.search('', {})
#> OK results

Profile.pagy_search('', {})
/Users/frexuz/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/pagy-6.4.3/lib/pagy/extras/meilisearch.rb:13:in `pagy_meilisearch': wrong number of arguments (given 2, expected 0..1) (ArgumentError)

il try making a debug file..

**Comment by Frexuz** [comment]
@ddnexus added debug file, same error
```
/Users/frexuz/.rbenv/versions/3.1.4/lib/ruby/gems/3.1.0/gems/pagy-6.4.3/lib/pagy/extras/meilisearch.rb:13:in `pagy_meilisearch': wrong number of arguments (given 2, expected 0..1) (ArgumentError)
	from pagy.rb:90:in `<main>'
```

**Comment by ddnexus** [comment]
> I don't see how thats related to application code?

Sorry, I was in autopilot. The reason is that the first filter that I had to implement in order to avoid wasting tons of time guessing other people code/intentions/dreams is, indeed, first checking for self contained code that can be reproduced and just asking for it if it's not, regardless the actual content. I hope you understand.

In your case, and despite the typo, the problem appears to be the argument syntax. I don't use `Meilisearch` myself and I am not the author of the extra, so I may not be the best person to comment on this. The `pagy_search` mechanism is derived from other similar extras that I wrote (basically it should have the same argument signature, store the arguments and pass them later to the intended target), but it looks like the `pagy_search` method is not expecting a hash, but optional keywords arguments:

https://github.com/ddnexus/pagy/blob/757d2e37af7885f258177c9d5d8cb2e9113f631b/lib/pagy/extras/meilisearch.rb#L13-L15

while `ms_search` (if the below one is the right one) uses a default empty hash:

https://github.com/meilisearch/meilisearch-rails/blob/29f59c88881b5a4b5a03d990f11d3aac220cd367/lib/meilisearch-rails.rb#L652-L656

That syntax is also enforced in the tests, so I am surprised to see the arguments of the `ms_search` in that form:

https://github.com/ddnexus/pagy/blob/757d2e37af7885f258177c9d5d8cb2e9113f631b/test/pagy/extras/meilisearch_test.rb#L18-L20

So the fix seems to be just aligning the arguments syntax of the `pagy_search` to the `ms_search` arguments.

@Frexuz as a temporary workaround, please, override the `pagy_search` in your model like:

```ruby
 def pagy_meilisearch(query, params = {}) 
   [self, query, params] 
 end 
```
and it should work.

I can push a fix. 

@molfar, @brunoocasali, could you please chime in for any unforeseen cases?

**Comment by ddnexus** [comment]
The workaround may work thanks to ruby, even with the double splatter here:

https://github.com/ddnexus/pagy/blob/757d2e37af7885f258177c9d5d8cb2e9113f631b/lib/pagy/extras/meilisearch.rb#L41

which should go away with the fix

**Comment by Frexuz** [comment]
Thanks @ddnexus the workaround works for now :)

**Comment by ddnexus** [comment]
@Frexuz Fix just released with pagy 6.4.4

**Comment by Frexuz** [comment]
Works. Thank you so much! <3
